src/main/java/carbonbot/Parser.java

```diff
@@ -0,0 +1,135 @@
+package carbonbot;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import carbonbot.command.AddCommand;
+import carbonbot.command.Command;
+import carbonbot.command.DeleteCommand;
+import carbonbot.command.ExitCommand;
+import carbonbot.command.FindCommand;
+import carbonbot.command.ListCommand;
+import carbonbot.command.MarkCommand;
+import carbonbot.task.Deadline;
+import carbonbot.task.Event;
+import carbonbot.task.Todo;
+
+/**
+ * The Parser class helps to parse user input strings to return the appropriate Command.
+ */
+public class Parser {
+    /**
+     * Parses the user input and returns the corresponding Command to be executed
+     *
+     * @param fullCommand The raw string the user has input
+     * @return The Command if the fullCommand string was valid
+     */
+    public static Command parse(String fullCommand) throws DukeException {
```

Comment:
Method body is abit long, perhaps you could split up into different smaller method?